src/api/user.ts

```diff
@@ -0,0 +1,92 @@
+import { Constants } from '../constants';
+import { fetchUser } from '../fetch';
+
+/* This function does the heavy lifting of processing data from Twitter API
+   and using it to create FixTweet's streamlined API responses */
+const populateUserProperties = async (
+  response: GraphQLUserResponse,
+  language: string | undefined
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+): Promise<APIUser> => {
+  const apiUser = {} as APIUser;
+
+  const user = response.data.user.result;
+  /* Populating a lot of the basics */
+  apiUser.url = `${Constants.TWITTER_ROOT}/${user.legacy.screen_name}`;
+  apiUser.id = user.rest_id;
+  apiUser.followers = user.legacy.followers_count;
+  apiUser.following = user.legacy.friends_count;
+  apiUser.likes = user.legacy.favourites_count;
+  apiUser.tweets = user.legacy.statuses_count;
+  apiUser.name = user.legacy.name;
+  apiUser.screen_name = user.legacy.screen_name;
+  apiUser.description = user.legacy.description;
+  apiUser.location = user.legacy.location;
+  apiUser.verified = user.legacy.verified;
+  apiUser.avatar_url = user.legacy.profile_image_url_https;
+  apiUser.joined = user.legacy.created_at;
+  if (user.legacy_extended_profile?.birthdate) {
+    const { birthdate } = user.legacy_extended_profile;
+    apiUser.birthday = {};
+    if (typeof birthdate.day === 'number') apiUser.birthday.day = birthdate.day;
+    if (typeof birthdate.month === 'number') apiUser.birthday.month = birthdate.month;
+    if (typeof birthdate.year === 'number') apiUser.birthday.year = birthdate.year;
+  }
+
+  return apiUser;
+};
+
+const writeDataPoint = (
```

Comment:
This should be relegated to its own file. I just didn't do it yet as this function is new (from within the last couple days). Either I or you can do that.

Also, the type of request should be stored in analytics engine as well so that aggregation can be done based off request types. This is just something I haven't done yet since I wrote this code just for status handler.

Reply:
Though, I did develop this analytics engine schema with statuses only in mind. It may be advisable to just remove this entirely for profile requests.